### ISSUE TITLE
bug in the handling of environmental variables containing the pipe "|" symbol in their value when using Docker

### ISSUE THREAD

**Issue by mriffle** [nextflow-io/nextflow#3690]
### Expected behavior and actual behavior

Observed behavior:

If I have this in my config file: `env.KEY1 = "apikey|blah"` and try to print the value of KEY1 in my process, I will get the following error:

```
/bin/bash: blah: command not found
.command.sh: line 3: KEY1 : unbound variable
```

It's clearly seeing the "|" and doing a *nix pipe to the non-existent "blah" command.

But, weirdly, if I have another environment variable defined before this one in the config file it works fine. E.g.:

```
env.KEY1 = "myenvvar"
env.KEY2 = "apikey|blah"
```

If I escape the "|" symbol using env.KEY1 = "apikey\\\\|blah", this works only when it's the only environment variable. If there is one defined before it (like just above), escaping it will result in a value of "apikey\\|blah" , which is a different value than the one it was given. It's as if the "|" is properly dealt with only if there is another env var defined before it.

Expected behavior: 

1. Having `env.KEY1 = "apikey|blah"` would not cause a bash error and would correctly set the environment variable.
2. The syntax for doing this shouldn't depend on whether another env variable was defined first.

### Steps to reproduce the problem

Using a docker executor, place this in the config file: `env.KEY1 = "apikey|blah"` and try to print the value of KEY1 in a process,

And using a docker executor, place this in a config file to see that having a previous env var defined results in different parsing behavior:

```
env.KEY1 = "myenvvar"
env.KEY2 = "apikey|blah"
```

### Program output 

```
/bin/bash: blah: command not found
.command.sh: line 3: KEY1 : unbound variable
```

### Environment 

* Nextflow version: 22.10.4
* Java version: openjdk version "11.0.17" 2022-10-18
* Operating system: [macOS, Linux, etc] Linux (Docker)
* Bash version: 4.2.46(2)-release

### Additional context

None

**Comment by pditommaso** [comment]
Likely it needs to be escaped here 

https://github.com/nextflow-io/nextflow/blob/448ccc14f3115fe0eeafc67757a713fd5c915011/modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy#L166-L172

**Comment by pditommaso** [comment]
I've tried the following and it worksfine 

```
process foo {
  debug true
  container 'ubuntu'

  '''
  echo "Hello $KEY2"
  '''
}

workflow {
  foo()
}
```

Output

```
N E X T F L O W  ~  version 23.02.1-edge
executor >  local (1)
[46/b855f0] process > foo [100%] 1 of 1 ✔
Hello apikey|blah
```

Closing this. Feel free to re-open if you have replicable test case

**Comment by mriffle** [comment]
Run it in docker

**Comment by pditommaso** [comment]
I did

**Comment by mriffle** [comment]
How did you set the env variable?

**Comment by pditommaso** [comment]
adding the following in the nextflow.config 

```
env.KEY1 = "myenvvar"
env.KEY2 = "apikey|blah"
```

**Comment by mriffle** [comment]
yes, that will work, but try it without the first line

**Comment by mriffle** [comment]
It's a very strange bug. 

If you have just:

```
env.KEY2 = "apikey|blah"
```

You will get the error.

But if you have:

```
env.KEY1 = "myenvvar"
env.KEY2 = "apikey|blah"
```

You will not get the error.

**Comment by pditommaso** [comment]
Ok, solved 👉 ecb7e0ee. Thanks for reportint it.
